src/nautilus_librarian/main.py

```diff
@@ -4,7 +4,7 @@
 import typer
 from typer.main import Typer
 
-from _version import __version__
+from nautilus_librarian._version import __version__
```

Comment:
I'm very new to python so I don't exactly understand what is happening here.

If I leave this `nautilus_librarian.` prefix in my code, it returns the default version number of 0.0.0. If I remove the prefix, it correctly shows the version number in my local version. The example pull request has the prefix.

I don't know enough about imports to understand why this is happening, so feel free to let me know if it should be different. The other referenced codebase had it prefixed, so it feels very possible that it's a local configuration vagary with my machine and that it should be prefixed.

Reply:
Try to execute:

```
pip install poetry-dynamic-versioning
poetry version
```

This time it should show you something like:

```
nautilus-librarian 0.2.1.post3.dev0+10871b5
```

What I understand is, once you install the `poetry-dynamic-versioning` plugin, `poetry version` gets the version from the git info instead of the `pyproject.toml`.

Reply:
If you run:

```
poetry run nautilus-librarian --version
Librarian Version: 0.0.0
```

You still get the version in the `pyproject.toml`.

If you build the dist:

```
$ poetry install && poetry build
Building nautilus-librarian (0.2.1.post3.dev0+10871b5)
  - Building sdist
  - Built nautilus-librarian-0.2.1.post3.dev0+10871b5.tar.gz
  - Building wheel
  - Built nautilus_librarian-0.2.1.post3.dev0+10871b5-py3-none-any.whl
```

And you install the package locally on a new terminal (using a generic Python environment):

```
pip install --user --force-reinstall dist/nautilus_librarian-0.2.1.post3.dev0+10871b5-py3-none-any.whl
```

you should get the right version:

```
nautilus-librarian --version
Librarian Version: 0.2.1.post2.dev0+b6c2ed0
```

because the wheel package contains the right version. In my case it's not working because there is an error during the installation.

```
ERROR: Error initializing plugin EntryPoint(name='macOS', value='keyring.backends.macOS', group='keyring.backends').
Traceback (most recent call last):
  File "/usr/share/python-wheels/resolvelib-0.5.4-py2.py3-none-any.whl/resolvelib/resolvers.py", line 171, in _merge_into_criterion
    crit = self.state.criteria[name]
KeyError: 'typer[all]'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/josecelano/.local/lib/python3.9/site-packages/keyring/backend.py", line 202, in _load_plugins
    init_func = ep.load()
  File "/usr/lib/python3.9/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 984, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'keyring.backends.macOS'
```

Ignoring that error, I think the way it works is:

The poetry-dynamic-versioning plugin changes the behavior of poetry in two different ways:

1. When you build the distribution package (`poetry build`) it changes the version placeholder in the `_version.py` file and also the version in the package metadata:

```
Metadata-Version: 2.1
Name: nautilus-librarian
Version: 0.2.1.post3.dev0+10871b5
```
2. When you call `poetry version` to get the package version it's going to call the plugin and read get the version from git, not from the `pyproject.toml`.

The point 2 is the one I'm discussing with @da2ce7. I have not read anything about it in the official documentation.

Reply:
@josecelano

From reading the code, it seems like ’poetry-dynamic-versioning patches’ the ’poetry’ main-command it's self.

This means from poetry, the ’pyproject.toml’ file always has the correct version:

```python
def _patch_poetry_command_run(run_mod) -> None:
    original_poetry_command_run = getattr(run_mod, "RunCommand").handle

    @functools.wraps(original_poetry_command_run)
    def alt_poetry_command_run(self, *args, **kwargs):
        # As of version 1.0.0b2, on Linux, the `poetry run` command
        # uses `os.execvp` function instead of spawning a new process.
        # This prevents the atexit `deactivate` hook to be invoked.
        # For this reason, we immediately call `deactivate` before
        # actually executing the run command.
        deactivate()
        return original_poetry_command_run(self, *args, **kwargs)

    getattr(run_mod, "RunCommand").handle = alt_poetry_command_run
```
https://github.com/mtkennerly/poetry-dynamic-versioning/blob/master/poetry_dynamic_versioning/__init__.py#L379

So the ’version’ sub-command isn't actually modified.

Reply:
hi @da2ce7 I asked the [question](https://github.com/mtkennerly/poetry-dynamic-versioning/issues/67#issuecomment-1023475983) in their repo and they confirmed the [`poetry version` command also changes its behaviour](https://github.com/mtkennerly/poetry-dynamic-versioning/issues/67#issuecomment-1023475983).